File: C++/find-center-of-star-graph.cpp
class Solution {
public:
    
    int findCenter(vector<vector<int>>& edges) {
        vector<int> nodes(edges.size() + 2);
        for (vector<int> &edge: edges) {
            nodes[edge[0]]++;
            nodes[edge[1]]++;
            if (nodes[edge[0]] > 1) {
                return edge[0];
            } else if (nodes[edge[1]] > 1) {
                return edge[1];
            }
        }
        return -1;
    }
};

File: Go/find-center-of-star-graph.go
func findCenter(edges [][]int) int {
    degree := map[int]int{}
    for _, e := range edges {
        a, b := e[0], e[1]
        degree[a]++
        if degree[a] > 1 {
            return a
        }
        degree[b]++
        if degree[b] > 1 {
            return b
        }        
    }
    return -1
}